### PR TITLE
feat: Adding extended splash screen 

### PR DESCRIPTION
**Pull request by nickrandolph**
Replaces #351

GitHub Issue (If applicable): #352 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the current behavior?

No easy way to extend the native splash screen to cover app loading (ie prevent black/white screen before app has chance to render first page)

## What is the new behavior?

ExtendedSplashScreen control can be set as root of application and it will load the native splashscreen as background (default style). Additional loading content (eg progressring) can be added via the LoadingContent property which is an overlay over the native splashscreen background

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by francoistanguay — APPROVED** [review]

File: src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj [hunk 1]
```diff
@@ -43,16 +43,21 @@
...
 	<ItemGroup>
+	  <UpToDateCheckInput Remove="Controls\ExtendedSplashScreen\ExtendedSplashScreen.xaml" />
 	  <UpToDateCheckInput Remove="Controls\LoadingView\LoadingView.xaml" />
 	  <UpToDateCheckInput Remove="Controls\SafeArea\SafeArea.xaml" />
 	</ItemGroup>
 
 	<ItemGroup>
+	  <Page Update="Controls\ExtendedSplashScreen\ExtendedSplashScreen.xaml">
+	    <SubType>Designer</SubType>
+	  </Page>
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj:60]: I feel like these None, UpToDateCheckInput, and Page items don't need to be here

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml [hunk 2]
```diff
@@ -42,7 +43,29 @@
 			</ResourceDictionary.MergedDictionaries>
 
 			<x:Double x:Key="DesktopAdaptiveThresholdWidth">720</x:Double>
-			
-		</ResourceDictionary>
+
+			<!-- Note: This style is defined here, rather than in a separate file (eg Styles/Application/Splashscreen.xaml)
+			because it depends on DefaultExtendedSplashScreen which is in ToolkitResources and resources need to be
+			merged so that it can be referenced - this seems to be an issue on some platforms -->
```
Comment (kazo0) [inline samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml:49]: whaaat, weird. This means we can't reference ResourceDictionaries that have references to any Material/Toolkit resources in subsequent files within the MergedDictionaries for App.xaml?

Like Styles/Button.xaml can't contain a style that is based on MaterialTextButtonStyle, say?

File: src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs [hunk 3]
```diff
@@ -137,5 +137,24 @@ private void UpdateVisualState()
 
 			VisualStateManager.GoToState(this, loadingState, IsLoaded);
 		}
+
+		#region DependencyProperty: Source
+
+		public static DependencyProperty SplashScreenProperty { get; } = DependencyProperty.Register(
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs:143]: I think I'm not understanding something, how come this needs to be a property on LoadingView and not directly on ExtendedSplashScreen? It looks like ExtendedSplashScreen is hiding this property behind its own SplashScreen property.

Was this initially here when SplashScreenPresenter was supposed to be a LoadingView attached property? I'm not seeing this attached property anymore.

File: src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs [hunk 4]
```diff
@@ -137,5 +137,24 @@ private void UpdateVisualState()
 
 			VisualStateManager.GoToState(this, loadingState, IsLoaded);
 		}
+
+		#region DependencyProperty: Source
```
Comment (kazo0) [inline src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs:141]: Should be `SplashScreen` instead of `Source`

Comment (nickrandolph) [reply src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs]: Splash screen property has been removed

File: src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs (hunk 3 above)
Comment (nickrandolph) [reply src/Uno.Toolkit.UI/Controls/LoadingView/LoadingView.cs]: Splash screen property has been removed

File: samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml (hunk 2 above)
Comment (nickrandolph) [reply samples/Uno.Toolkit.Samples/Uno.Toolkit.Samples.Shared/App.xaml:49]: That does seem to be an issue - let's verify

File: src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj (hunk 1 above)
Comment (nickrandolph) [reply src/Uno.Toolkit.UI/Uno.Toolkit.UI.csproj]: I've removed these lines

**Pull request merged**